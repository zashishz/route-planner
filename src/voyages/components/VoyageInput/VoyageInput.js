import React from "react";

import "./VoyageInput.css";

const VoyageInput = ({ label, refNode, onChange, onClick }) => (
  <div className="form-input">
    <label>{label}</label>
    <div className="form-input-controls">
      <input
        placeholder={label}
        type="text"
        ref={refNode}
        onChange={onChange}
      />
      <button onClick={onClick}>X</button>
    </div>
  </div>
);

export default VoyageInput;
