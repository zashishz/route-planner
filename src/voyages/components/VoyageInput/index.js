import VoyageInput from "./VoyageInput";

export { VoyageInput };
