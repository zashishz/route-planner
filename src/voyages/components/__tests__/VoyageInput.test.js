import React from "react";
import renderer from "react-test-renderer";

import { VoyageInput } from "../VoyageInput";

describe("Render VoyageInput correctly", () => {
  it("render correctly", () => {
    const tree = renderer.create(<VoyageInput />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
