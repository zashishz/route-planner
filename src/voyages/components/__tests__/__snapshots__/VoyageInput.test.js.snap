// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render VoyageInput correctly render correctly 1`] = `
<div
  className="form-input"
>
  <label />
  <div
    className="form-input-controls"
  >
    <input
      type="text"
    />
    <button>
      X
    </button>
  </div>
</div>
`;
